pylint/checkers/design_analysis.py

```diff
@@ -95,16 +95,17 @@
 DATACLASSES_DECORATORS = frozenset({"dataclass", "attrs"})
 DATACLASS_IMPORT = "dataclasses"
 TYPING_NAMEDTUPLE = "typing.NamedTuple"
+TYPING_TYPEDDICT = "typing.TypedDict"
 
 
 def _is_exempt_from_public_methods(node: astroid.ClassDef) -> bool:
     """Check if a class is exempt from too-few-public-methods"""
 
-    # If it's a typing.Namedtuple or an Enum
+    # If it's a typing.Namedtuple, typing.TypedDict or an Enum
     for ancestor in node.ancestors():
         if ancestor.name == "Enum" and ancestor.root().name == "enum":
             return True
-        if ancestor.qname() == TYPING_NAMEDTUPLE:
+        if ancestor.qname() in (TYPING_NAMEDTUPLE, TYPING_TYPEDDICT):
```

Comment:
Shouldn't we create a allow list for class that can have a few public methods in configuration ? The entirety of django models and form classes would fit for example. This is outside the scope of this MR of course but this is the refactor that comes to mind seeing this.

Reply:
Sounds reasonable

Reply:
Should I open an issue or could you do it in this MR ? I think this it merge-able as it is.

Reply:
I think that should be done separately. The MR is just a bug fix whereas that would be a new feature.

Reply:
Opened #4197 for it.